capreolus/task/rank.py

```diff
@@ -12,7 +12,11 @@
 class RankTask(Task):
     module_name = "rank"
     requires_random_seed = False
-    config_spec = [ConfigOption("optimize", "map", "metric to maximize on the dev set"), ConfigOption("filter", False)]
+    config_spec = [
+        ConfigOption("filter", False),
+        ConfigOption("optimize", "map", "metric to maximize on the dev set"),
+        ConfigOption("metrics", "default", "metrics reported for evaluation", value_type="strlist"),
+    ]
     config_keys_not_in_path = ["optimize"]  # only used for choosing best result; does not affect search()
```

Comment:
We can add `metrics` to `config_keys_not_in_path` since it won't affect anything in `get_results_path()`